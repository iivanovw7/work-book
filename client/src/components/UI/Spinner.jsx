import PropTypes from 'prop-types';
import React from 'react';
import styled, { ThemeProvider } from 'styled-components';
import { spin } from '../../styles';
import { spinnerBackground, textLinkColorLighten } from '../../theme';
/* eslint react/require-default-props: 0 */

const StyledWrapper = styled.div`
	display: flex;
  flex-direction: column;
  align-items: center;
  justify-content: center;
  width: 100%;
  height: 100%;
  
  div {
    margin: ${props => props.margin};
    padding: ${props => props.padding};
  }
`;

const StyledSpinner = styled.div`
  ${spin}
  display: inline-block;
  width: 50px;
  height: 50px;
  border-radius: 50%;
  border: 3px solid ${spinnerBackground};
  border-top-color: ${textLinkColorLighten};

`;

const Spinner = (props) => {
  const { theme, margin, padding } = props;

  return (
    <ThemeProvider theme={{ mode: theme }}>
      <StyledWrapper margin={margin} padding={padding}>
        <StyledSpinner variant="secondary" />
      </StyledWrapper>
    </ThemeProvider>
  );
};

export default Spinner;

StyledSpinner.propTypes = {
  variant: PropTypes.oneOf(['primary', 'secondary'])
};

StyledSpinner.defaultProps = {
  variant: 'primary',
  mgTop: '0',
  padding: '0'
};

Spinner.propTypes = {
  theme: PropTypes.string.isRequired,
  padding: PropTypes.string,
  margin: PropTypes.string
};
