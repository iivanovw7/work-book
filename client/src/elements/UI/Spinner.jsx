import PropTypes from 'prop-types';
import React from 'react';
import styled, { ThemeProvider } from 'styled-components';
import { btnBackground } from '../../theme';

const StyledWrapper = styled.div`
	display: flex;
  flex-direction: column;
  align-items: center;
  justify-content: center;
  width: 100%;
  height: 100%;
  
  div {
    margin-top: ${props => props.mgTop}em;
  }
`;

const StyledSpinner = styled.div`
  display: inline-block;
  width: 50px;
  height: 50px;
  border: 3px solid rgba(0, 0, 0, 0.3);
  border-radius: 50%;
  border-top-color: ${btnBackground};
  animation: spin 1s ease-in-out infinite;
  -webkit-animation: spin 1s ease-in-out infinite;

	@keyframes spin {
		0% {
			transform: rotate(0deg);
    	-webkit-transform: rotate(0deg);
		}
		
		100% {
			transform: rotate(360deg);
    	-webkit-transform: rotate(360deg);
		}
	}
 

	@-webkit-keyframes spin {
		0% {
			transform: rotate(0deg);
    	-webkit-transform: rotate(0deg);
		}
		
		100% {
			transform: rotate(360deg);
    	-webkit-transform: rotate(360deg);
    }
  }
`;

const Spinner = (props) => {
  const { theme, mgTop } = props;

  return (
    <ThemeProvider theme={{ mode: theme }}>
      <StyledWrapper mgTop={mgTop}>
        <StyledSpinner variant="primary" />
      </StyledWrapper>
    </ThemeProvider>
  );
};

export default Spinner;

Spinner.propTypes = {
  theme: PropTypes.string.isRequired,
  mgTop: PropTypes.string
};

StyledSpinner.propTypes = {
  variant: PropTypes.oneOf(['primary', 'secondary'])
};

StyledSpinner.defaultProps = {
  variant: 'primary',
  mgTop: 0
};
