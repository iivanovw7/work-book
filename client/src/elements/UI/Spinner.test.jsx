import 'cross-fetch/polyfill';
import 'jsdom-global/register';
import React from 'react';
import { mount } from 'enzyme';
import chalk from 'chalk';
import Spinner from './Spinner';

jest.mock('../../config/apiURL', () => () => '/api');

describe(`Testing [${chalk.yellow('Spinner')}] component`, () => {
  afterAll(() => {
    jest.clearAllMocks();
  });

  const props = {
    theme: 'dark',
    mgTop: 1
  };

  const Composition = () => (
    <Spinner userAccess {...props} />
  );

  it('Should match snapshot', () => {
    const component = mount(<Composition />);

    expect(component).toMatchSnapshot();
  });

  it('Should render without an error', () => {
    const component = mount(<Composition />);
    expect(component.length).toEqual(1);
  });

  it('Should render with correct props', () => {
    const component = mount(<Composition />);

    expect(component.find(Spinner).prop('mgTop')).toBe(1);
    expect(component.find(Spinner).prop('theme')).toBe('dark');
  });
});
