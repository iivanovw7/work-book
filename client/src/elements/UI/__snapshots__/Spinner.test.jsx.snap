// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing Spinner component Should match snapshot 1`] = `ReactWrapper {}`;
